### PR TITLE
Reader Stats: pageViewForPost should be safe

### PR DESCRIPTION
**Pull request by samouri**
currently breaking on site_ID: 1.

**testing instructions**
go to: http://calypso.localhost:3000/read/feeds/77164643/posts/1724351807

### CODE REVIEW

**Comment by dmsnell** [comment]
Does it make sense to you why we should be getting `site_ID === 1`?

**Comment by dmsnell** [comment]
Should we not prevent the call to `markAllSeen()` in `attemptToSendPageView()`? I don't mind leaving the other check in but I think the problem is that we're trying to send that call in a place where we know the site may not be valid. Or even in `markPostSeen()`?

**Comment by dmsnell** [comment]
I feel like this patch would be a tighter match to the cause of the problem

```
diff --git a/client/blocks/reader-full-post/index.jsx b/client/blocks/reader-full-post/index.jsx
index ecaec4bd0a..c67bf1f175 100644
--- a/client/blocks/reader-full-post/index.jsx
+++ b/client/blocks/reader-full-post/index.jsx
@@ -235,7 +235,7 @@ export class FullPostView extends React.Component {
        attemptToSendPageView = () => {
                const { post, site } = this.props;

-               if ( post && post._state !== 'pending' && site && site.ID && ! this.hasSentPageView ) {
+               if ( post && post._state !== 'pending' && site && site.ID && ! site.is_error && ! this.hasSentPageView ) {
                        this.props.markPostSeen( post, site );
                        this.hasSentPageView = true;
                }
```

**Comment by samouri** [comment]
throw that in there! lets do both

File: client/blocks/reader-full-post/index.jsx [hunk 1]
```diff
@@ -235,7 +235,14 @@ export class FullPostView extends React.Component {
 	attemptToSendPageView = () => {
 		const { post, site } = this.props;
 
-		if ( post && post._state !== 'pending' && site && site.ID && ! this.hasSentPageView ) {
```
Comment (samouri) [inline client/blocks/reader-full-post/index.jsx]: this is one ugly check. 
I could see us adding some common utils later for: `isValidPost` and `isValidSite`

**Comment by samouri** [comment]
Thanks @dmsnell! Changes LGTM . 
Testing now

File: client/blocks/reader-full-post/index.jsx [hunk 2]
```diff
@@ -235,7 +235,14 @@ export class FullPostView extends React.Component {
...
+		if (
+			post &&
+			post._state !== 'pending' &&
+			site &&
+			site.ID &&
+			! site.is_error &&
+			! this.hasSentPageView
```
Comment (gwwar) [inline client/blocks/reader-full-post/index.jsx]: Probably worth adding a few comments too to explain the more unusual checks. Out of context I'm not sure folks realize there's a reader site for example that may have an `is_error` prop.

**Pull request merged**